maro/rl_v3/policy_trainer/abs_trainer.py

```diff
@@ -19,12 +21,19 @@ def __init__(self, name: str, device: str = None) -> None:
 
         print(f"Creating trainer {self.__class__.__name__} {name} on device {self._device}")
 
+        self._task_queue_client = Optional[TaskQueueClient]
+
     @property
     def name(self) -> str:
         return self._name
 
+    def train_step(self, data_parallel: bool = False) -> None:
+        if data_parallel:
+            assert self._task_queue_client is not None
```

Comment:
This can be done during initializing `data_parallel`, instead of checking everytime.
So the `_train_step_impl` can remain as `train_step`.

Reply:
I think the assertion in `train_step` is still necessary, because `self._task_queue_client` is not initialized in the `__init__` function. Therefore, `self._task_queue_client` could be `None` even if `self._data_parallel` is `True`.